buku

```diff
@@ -6282,6 +6308,10 @@ POSITIONAL ARGUMENTS:
 
     update_search_results = False
     if search_results:
+        if args.random and args.random < len(search_results):
+            search_results = bdb._sort(random.sample(search_results, args.random), order)
+        single_record = args.random == 1  # matching print_rec() behaviour
```

Comment:
Currently `single_record` is tied to `--random`.

(This can be changed to always using single record mode when only one is found.)